pkg/concurrent/concurrent.go

```diff
@@ -17,11 +17,15 @@ limitations under the License.
 package concurrent
 
 import (
+	"errors"
 	"sync"
+	"time"
 
 	"sigs.k8s.io/kind/pkg/util"
 )
 
+const timeout = 60 * time.Second
```

Comment:
Should we start with a higher value?

Reply:
it will mask the problem and throw the error after a minute.
were we able to figure out why the hangs happen?

is `log.Debugf("Untarring file %v to %v", f.Name, dir)` consistently showing the same file?

Reply:
1. The problem will always be masked, I mean, the only difference with the current situation is that it won´t hang forever, only for the time we want it to hang. 
It will fail with an error saying that it timed out in one concurrent function and we will need to track it down using the logs like we are doing now. I think that´s a good safeguard at this moment, simple and effective. If we continue to hit this kind of problems I think we should start to use other options like `context`, but seems to overkill at this point, however, I´m not an expert in this topic.

2. It prints the name of the file it´s untarring, i.e.
```
DEBU[14:25:38] Untarring file ./pods/kube-system_kindnet-6244g_65fcc7c7-6eab-4024-8323-ec90df9d3582/kindnet-cni/0.log to /tmp/230018879/kind-control-plane
DEBU[14:25:38] Untarring file ./containers/busybox_default_busybox-e31c112c91afda6d8e51cb412100e2a555e97dfe6c4cd8d3550a052d6cea0c5c.log to /tmp/230018879/kind-worker2
DEBU[14:25:38] Untarring file ./containers/nginx-deployment-65bbbdfcd-v6m4b_default_nginx-a77eb9aa92d867c4bfee267ed2bcb822bee022f233420740097cde22412cd601.log to /tmp/230018879/kind-worker2
```

Reply:
> The problem will always be masked

a timeout in general is OK to have, but i'm missing context on the actual problem, is it related to mount points? why does it hang?

Reply:
ups, sorry, in this case, [is about the `tar` problem, ](https://github.com/kubernetes-sigs/kind/pull/606#issuecomment-501664744 ) but don't know if it can happen with others functions that we execute concurrently.
 If they hang, the concurrency functions never time out

Reply:
i guess i was trying to understand why the tar operation actually hangs - e.g. is something about the `io.Reader` or about the logic in `archive/tar`.

technically `time.Sleep(1000 * time.Second)` will act as a delay there.

Reply:
`time.Sleep(1000 * time.Second)` was an example that I've used to see the effects of the `untar`  function hanging.
If I understood correctly the current issue with tar was fixed  with #608 moving `err2 := <-errChan` after the `if err != nil {	`

```
	go func() {
		errChan <- readerFunc(pr)
	}()		
	err = cmd.Run()		
	err2 := <-errChan	
	if err != nil {		
		return err
	}		
``` 

I think that what happened is that the `cmd.Run()` failed but it never returns because is waiting for `err2 := <-errChan`, that;s waiting for the readerFunc()

Reply:
this is known, but also doesn't occur in practical contexts currently.

timeouts are hard to get right. what if there are lots of logs?
ideally we want to detect actual hangs rather than blind timeouts.

Reply:
yeah, eventually, the job will timeout too, so we should decide which timeout is more useful
With this approach we will have an error saying `Coalesce: Timeout waiting for wait group"` , we can set a long timeout, 5 mins per example to be safe, just something lower than the prow job timout